src/span_context.js

```diff
@@ -0,0 +1,163 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import * as constants from './constants.js';
+import Utils from './util.js';
+import Int64 from 'node-int64';
+
+export default class SpanContext {
+    _traceId: any;
+    _spanId: any;
+    _parentId: any;
+    _flags: number;
+    _baggage: any;
+    _baggageHeaderCache: any;
+
+    constructor(traceId: any,
+                spanId: any,
+                parentId: any,
+                flags: number) {
+        this._traceId = traceId;
+        this._spanId = spanId;
+        this._parentId = parentId;
+        this._flags = flags;
+        this._baggage = {};
+        this._baggageHeaderCache = {};
+    }
+
+    get traceId(): any {
+        return this._traceId;
+    }
+
+    get spanId(): any {
+        return this._spanId;
+    }
+
+    get parentId(): any {
+        return this._parentId;
+    }
+
+    get flags(): number {
+        return this._flags;
+    }
+
+    get baggage(): any {
+        return this._baggage;
+    }
+
+    /**
+     * Returns a normalize key.
+     *
+     * @param {string} key - The key to be normalized for a particular baggage value.
+     * @return {string} - The normalized key (lower cased and underscores replaced, along with dashes.)
+     **/
+    // todo(oibe) moved to span in newest opentracing version
+    _normalizeBaggageKey(key: string) {
+        if (key in this._baggageHeaderCache) {
+            return this._baggageHeaderCache[key];
+        }
+
+        // This stackoverflow claims this approach is slower than regex
+        // http://stackoverflow.com/questions/1144783/replacing-all-occurrences-of-a-string-in-javascript
+        let normalizedKey: string = key.toLowerCase().split('_').join('-');
+
+        if (Object.keys(this._baggageHeaderCache).length < 100) {
+            this._baggageHeaderCache[key] = normalizedKey;
+        }
+
+        return normalizedKey;
+    }
+
+    /**
+     * Sets a baggage value with an associated key.
+     *
+     * @param {string} key - The baggage key.
+     * @param {string} value - The baggage value.
+     **/
+    // todo(oibe) moved to span in newest opentracing version
+    setBaggageItem(key: string, value: string): void {
+        let normalizedKey = this._normalizeBaggageKey(key);
+        this._baggage[normalizedKey] = value;
+    }
+
+    /**
+     * Gets a baggage value with an associated key.
+     *
+     * @param {string} key - The baggage key.
+     * @return {string} value - The baggage value.
+     **/
+    // todo(oibe) moved to span in newest opentracing version
+    getBaggageItem(key: string): string {
+        let normalizedKey = this._normalizeBaggageKey(key);
+        return this._baggage[normalizedKey];
+    }
+
+    /**
+     * @return {boolean} - returns whether or not this span context was sampled.
+     **/
+    IsSampled() {
+        return !!(this.flags & constants.SAMPLED_MASK);
```

Comment:
This is not double negation, but the operator to cast to a boolean instead of returning a number.